### PR TITLE
access_users: Add ZeroTrust Users & Seats functions.

### LINKED ISSUE #1425: Zerotrust / Access User endpoints.
### Current cloudflare-go version

v0.80.0

### Description

Addition of the User methods documented at https://developers.cloudflare.com/api/operations/zero-trust-users-get-users 

### Use cases

The following API endpoint would be used as part of JML associated with the licence management and incident response processes related to reviewing & revoking cloudflare access / zero trust usage. 

### Potential cloudflare-go usage

```go
# Example useage

api := yourpackage.NewAPI("API-KEY")

// Define a resource container for the organization level.
orgResource := &yourpackage.ResourceContainer{
    Level:      "organizations",
    Identifier: "ORG-ID",
}

// Define parameters for listing access groups.
listParams := yourpackage.ListAccessGroupsParams{
    PerPage: 50, // Number of results per page.
    Page:    1,  // Page number.
}

// Retrieve a list of access users for the access organization.
accessUsers, _, err := api.GetAccessUsers(context.Background(), orgResource, listParams)
if err != nil {
    fmt.Printf("Error fetching access users: %v\n", err)
    return
}

# Example implementation based on ListAccessGroups

// GetAccessUsers returns all users access within a Cloudflare Access Organisation.
//
// Access API Reference: https://developers.cloudflare.com/api/operations/zero-trust-users-get-users
func (api *API) GetAccessUsers(ctx context.Context, rc *ResourceContainer, params ListAccessGroupsParams) ([]AccessGroup, *ResultInfo, error) {
	baseURL := fmt.Sprintf("/%s/%s/access/users", rc.Level, rc.Identifier)

	autoPaginate := true
	if params.PerPage >= 1 || params.Page >= 1 {
		autoPaginate = false
	}

	if params.PerPage < 1 {
		params.PerPage = 25
	}

	if params.Page < 1 {
		params.Page = 1
	}

	var acccessUsers []AccessUser
	var r AccessGroupListResponse

	for {
		uri := buildURI(baseURL, params)
		res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
		if err != nil {
			return []AccessGroup{}, &ResultInfo{}, fmt.Errorf("%s: %w", errMakeRequestError, err)
		}

		err = json.Unmarshal(res, &r)
		if err != nil {
			return []AccessGroup{}, &ResultInfo{}, fmt.Errorf("%s: %w", errUnmarshalError, err)
		}
		acccessUsers = append(acccessUsers, r.Result...)
		params.ResultInfo = r.ResultInfo.Next()
		if params.ResultInfo.Done() || !autoPaginate {
			break
		}
	}

	return acccessUsers, &r.ResultInfo, nil
}
```

### References

_No response_

### PR DESCRIPTION
**Pull request by gavinelder**
## Description

Closes #1425

The following adds the Cloudflare Zerotrust / Access endpoints for user & seat management. 

The following is a rough draft with comments for feedback. 

## Has your change been tested?

Basic API calls locally against relevant endpoints to ensure a proper response is given. 

## Types of changes

What sort of change does your code introduce/modify?

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] This change is using publicly documented in [cloudflare/api-schemas](https://github.com/cloudflare/api-schemas) 
      and relies on stable APIs.

[1]: https://help.github.com/articles/closing-issues-using-keywords/

### CODE REVIEW

File: access_seats.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+// UpdateAccessUserSeat updates a Access User Seat.
+// UpdateAccessUserSeat Removes a user from a Zero Trust seat when both access_seat and gateway_seat are set to false.
+//
+// Access API reference: https://developers.cloudflare.com/api/operations/zero-trust-seats-update-a-user-seat
+func (api *API) UpdateAccessUserSeat(ctx context.Context, rc *ResourceContainer, params UpdateAccessUserSeatParams) (AccessUserSeat, error) {
+	if params.SeatUID == "" {
+		return AccessUserSeat{}, errMissingAccessSeatUID
+	}
```
Comment (gavinelder) [inline access_seats.go]: Noticed on similar functions where there is a required parameter is needed there are in-function checks to ensure these are appropriately set. 

With regards to having the response in-line or abstracted out to a variable I have seen both examples in use, opted for abstraction here.

File: access_seats.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	Errors     []string       `json:"errors"`
+	Messages   []string       `json:"messages"`
+	Result     AccessUserSeat `json:"result"`
+	ResultInfo ResultInfo     `json:"result_info"`
+}
+
+// UpdateAccessUserSeat updates a Access User Seat.
+// UpdateAccessUserSeat Removes a user from a Zero Trust seat when both access_seat and gateway_seat are set to false.
+//
+// Access API reference: https://developers.cloudflare.com/api/operations/zero-trust-seats-update-a-user-seat
```
Comment (gavinelder) [inline access_seats.go]: Mirrored the comment structure of `account_roles.go` but happy to structure this as needed.

File: access_seats_test.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+func TestUpdateAccessUserSeatWithMissingUID(t *testing.T) {
+	setup()
+	defer teardown()
+
+	_, err := client.UpdateAccessUserSeat(context.Background(), testAccountRC, UpdateAccessUserSeatParams{})
+	assert.EqualError(t, err, "missing required access seat UID")
+
+	_, err = client.UpdateAccessUserSeat(context.Background(), testZoneRC, UpdateAccessUserSeatParams{})
+	assert.EqualError(t, err, "missing required access seat UID")
```
Comment (gavinelder) [inline access_seats_test.go]: I kind of feel that if the zoneRC is used it should error but I could not see evidence of this pattern.

File: access_user.go [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+	"fmt"
+	"net/http"
+
+	"github.com/goccy/go-json"
+)
+
+/*---------------------*/
+// GetAccessUserActiveSessions
+//
+/*---------------------*/
```
Comment (gavinelder) [inline access_user.go]: Comments just for logical separation will be removed during cleanup

File: access_user.go [hunk 5]
```diff
@@ -0,0 +1,382 @@
...
+}
+
+type AccessUserSingleActiveSessionGeo struct {
+	Country string `json:"country"`
+}
+
+type AccessUserSingleActiveSessionIDP struct {
+	ID   string `json:"id"`
+	Type string `json:"type"`
+}
```
Comment (gavinelder) [inline access_user.go]: There are instances around device posture checks where types are duplicated , these will be simplified , names are over-verbose for now.

**Review by gavinelder — COMMENTED** [review]
Feel free to add relevant comments . 

Right now as far as testing goes, outside of basic response unmarshelling I am unsure what you would like to see.

File: access_seats_test.go (hunk 3 above)
Comment (jacobbednarz) [reply access_seats_test.go]: example: https://github.com/cloudflare/cloudflare-go/blob/eff3c6531617a30c1bc1d07628dab365673a2d30/waiting_room.go#L562-L564

File: access_seats.go (hunk 2 above)
Comment (jacobbednarz) [inline access_seats.go]:
```suggestion
// API documentation: https://developers.cloudflare.com/api/operations/zero-trust-seats-update-a-user-seat
```

File: access_seats.go [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+type UpdateAccessUserSeatResponse struct {
+	Success    bool           `json:"success"`
+	Errors     []string       `json:"errors"`
+	Messages   []string       `json:"messages"`
+	Result     AccessUserSeat `json:"result"`
+	ResultInfo ResultInfo     `json:"result_info"`
+}
+
+// UpdateAccessUserSeat updates a Access User Seat.
+// UpdateAccessUserSeat Removes a user from a Zero Trust seat when both access_seat and gateway_seat are set to false.
```
Comment (jacobbednarz) [inline access_seats.go]:
```suggestion
// UpdateAccessUserSeat updates a Access User Seat.
```

File: access_seats.go [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+type UpdateAccessUserSeatParams struct {
+	SeatUID     string `json:"seat_uid"`
+	AccessSeat  bool   `json:"access_seat"`
+	GatewaySeat bool   `json:"gateway_seat"`
+}
+
+type UpdateAccessUserSeatResponse struct {
+	Success    bool           `json:"success"`
+	Errors     []string       `json:"errors"`
+	Messages   []string       `json:"messages"`
```
Comment (jacobbednarz) [inline access_seats.go]:
```suggestion
	Response
```

**Comment by gavinelder** [comment]
@jacobbednarz thanks for the review, this should hopefully be in a better state now.

File: access_seats.go [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+// AccessUpdateAccessUserSeatResult represents a Access User Seat.
+type AccessUpdateAccessUserSeatResult struct {
+	AccessSeat  bool       `json:"access_seat"`
+	CreatedAt   *time.Time `json:"created_at"`
+	UpdatedAt   *time.Time `json:"updated_at"`
+	GatewaySeat bool       `json:"gateway_seat"`
+	SeatUID     string     `json:"seat_uid,omitempty"`
+}
+
+// UpdateAccessUserSeatParams
```
Comment (jacobbednarz) [inline access_seats.go]:
```suggestion
```

**Comment by jacobbednarz** [comment]
thanks for this one @gavinelder 🏆

**Pull request merged**